### ISSUE TITLE
feat request: Batch inference with outlines applying the same regex

### ISSUE THREAD

**Issue by alonsosilvaallende** [dottxt-ai/outlines#1234]
I'm trying to make batch inference with outlines when I'm applying the same regex to a whole dataset.
I noticed the following comment in the logits processor:
https://github.com/dottxt-ai/outlines/blob/dc31b9b4027e71bd69f213bd3ab7d98b2f0a1529/outlines/processors/base_logits_processor.py#L47-L48
Why is it the case? Is it some hard constraint or can it be relaxed when applying the same regex?

**Comment by rlouf** [comment]
Does it work now?
